src/blocks/transformers/jq.ts

```diff
@@ -66,13 +68,28 @@ export class JQTransformer extends Transformer {
       // eslint-disable-next-line @typescript-eslint/return-await -- Type is `any`, it throws the rule off
       return await jq.promised.json(input, filter);
     } catch (error) {
-      if (!isErrorObject(error) || !error.message.includes("compile error")) {
+      // The message length check is there because the JQ error message sometimes is cut and if it is we try to parse the stacktrace
+      // See https://github.com/pixiebrix/pixiebrix-extension/issues/3216
+      if (
+        !isErrorObject(error) ||
+        (error.message.length > 13 && !error.message.includes("compile error"))
+      ) {
         throw error;
       }
 
-      const message = error.stack.includes("unexpected $end")
-        ? "Unexpected end of jq filter, are you missing a parentheses, brace, and/or quote mark?"
-        : "Invalid jq filter, see error log for details";
+      let message: string;
+      if (error.stack.includes("unexpected $end")) {
+        message =
+          "Unexpected end of jq filter, are you missing a parentheses, brace, and/or quote mark?";
+      } else {
+        const jqMessageFromStack = jqStacktraceRegexp.exec(error.stack)?.groups
```

Comment:
You can write it this way to avoid the nested condition. Prettier will likely improve the wrapping further

```suggestion
        message = jqStacktraceRegexp.exec(error.stack)?.groups?.message?.trim()
	        ?? "Invalid jq filter, see error log for details"
```